### PR TITLE
Recover the `traceback` from `PyErr` and use `PyBytes` as input

### PR DESCRIPTION
**Pull request by haixuanTao**
## Additions:
- Recover the traceback to see on which line Python failed.
- Use Pybytes as input to make Python input type `bytes` instead of a List of `u8`.

### CODE REVIEW

File: binaries/runtime/src/operator/python.rs [hunk 1]
```diff
@@ -47,17 +47,29 @@ pub fn spawn(
...
-        let module = py
-            .import(module_name)
-            .wrap_err("failed to import Python module")?;
+        let module = py.import(module_name).or_else(|err: pyo3::PyErr| {
+            let trace = err
+                .traceback(py)
+                .expect("PyError should have a traceback")
+                .format()
+                .wrap_err("Traceback could not be formatted")?;
+            Err(eyre!("{trace}"))
```
Comment (phil-opp) [inline binaries/runtime/src/operator/python.rs:56]: According to https://docs.rs/pyo3/0.16.5/pyo3/types/struct.PyTraceback.html#method.format , the traceback does not contain the actual error. So we should probably do somthing like this:

```suggestion
            Err(eyre!("{err}\n{trace}"))
```

Or

```suggestion
            Err(eyre!("{trace}{err}"))
```

File: binaries/runtime/src/operator/python.rs [hunk 2]
```diff
@@ -47,17 +47,29 @@ pub fn spawn(
...
         let operator = py
             .eval("Operator()", None, Some(locals))
-            .wrap_err("failed to create Operator instance")?;
+            .or_else(|err: pyo3::PyErr| {
+                let trace = err
+                    .traceback(py)
+                    .expect("PyError should have a traceback")
+                    .format()
+                    .wrap_err("Traceback could not be formatted")?;
+                Err(eyre!("{trace}"))
```
Comment (phil-opp) [inline binaries/runtime/src/operator/python.rs:71]: Looks like this part is duplicated a few times in this file. Perhaps we can factor it out into a separate `convert_py_err` method? To me the ideal solution would be if we could do just `.map_err(convert_py_err)` afterwards.

File: binaries/runtime/src/operator/python.rs (hunk 1 above)
Comment (haixuanTao) [reply binaries/runtime/src/operator/python.rs:56]: Indeed! Good catch! I went for `{trace}\n{err}` as it feels closer to the natural interpreter.

File: binaries/runtime/src/operator/python.rs (hunk 2 above)
Comment (haixuanTao) [reply binaries/runtime/src/operator/python.rs]: Yep! so I wasn't sure I could recursively get the Python GIL, but you actually can. I called the function `traceback` in accordance to its functionality.

**Review by phil-opp — APPROVED** [review]

**Pull request merged**